qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py

```diff
@@ -112,13 +112,18 @@ def _substitution_checks(self, dag, old_run, new_circ, basis, qubit):
         # if we're outside of the basis set, we're obligated to logically decompose.
         # if we're outside of the set of gates for which we have physical definitions,
         #    then we _try_ to decompose, using the results if we see improvement.
+        new_error = _error(new_circ, self._target, qubit)
+        if isinstance(new_error, tuple):
+            error_rate = new_error[0]
+        else:
+            error_rate = new_error
         return (
             uncalibrated_and_not_basis_p
             or (
                 uncalibrated_p
-                and _error(new_circ, self._target, qubit) < _error(old_run, self._target, qubit)
+                and new_error < _error(old_run, self._target, qubit)
             )
-            or np.isclose(_error(new_circ, self._target, qubit), 0)
+            or np.isclose(error_rate, 0)
```

Comment:
This check feels weird, because before this PR the last line could only have triggered if the error rate was "close to" 0, but not _actually_ 0 - in that case, it would have been replaced by `-100 + len(circuit)`, so not be close any more.  I'm not sure if that was intended - the behaviour it'll have after this PR (anything "close to" zero triggers the replacement) feels more natural to me.